model/src/main/kotlin/ScannerRun.kt

```diff
@@ -29,6 +29,7 @@ import org.ossreviewtoolkit.model.config.ScannerConfiguration
 import org.ossreviewtoolkit.model.utils.FileListSortedSetConverter
 import org.ossreviewtoolkit.model.utils.ProvenanceResolutionResultSortedSetConverter
 import org.ossreviewtoolkit.model.utils.ScanResultSortedSetConverter
+import org.ossreviewtoolkit.model.utils.ScannersMapConverter
```

Comment:
commit-message: This commit fixes a more general problem than what is described in the commit message. 

The scanner may used different set of scanners for scanning different packages. The same holds for projects. `ScannerRun` does not contain that information, and `ScannerRun.getScanResultsById()` just returns all scans matching the packages (nested) provenance, disregarding which scanners are to be used for the particular package. Extend the data model to contain that information and use it to adjust the internals of `getScanResultsById()` to adhere to the set of scanners to be used for the particular project or package.


----

Just as a hint: Which scanner to use for scanning a project / package depends on (haven't double checked):
- `--skip-excluded`
- `--skip-concluded`
- `--scanners`
- `--project-scanners

Reply:
I don't understand the last sentence of your comment.

Reply:
I suppose you mean the sentence below the horizontal line, I meant to list the things which can influence which scanner are to be used for a package / project.

Reply:
Understand. Maybe also `--package-types` ?